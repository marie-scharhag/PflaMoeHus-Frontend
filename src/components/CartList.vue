<template>
    <div class= "cartlist">
        <ul class="list">
         <CartListObject :product="pr" v-for="pr  in productList" :key="pr.id" />
        </ul>
        
        <div class="inTotalDiv">
            <span class="inTotal">Gesamtsumme: {{inTotal}} â‚¬</span>
        </div>
    </div>
</template>
<script lang ="ts">
import CartListObject from "../components/CartListObject.vue"
import {computed} from 'vue'; 
import {useCartStore} from '../service/CartStore'

export default {
  components: {
    CartListObject,
  },
  setup() {
    const { list, totalPrice } = useCartStore();
    /**
     * the total price of all products in cart
     */
    const inTotal = computed(() => {
      return totalPrice();
    });
    /**
     * all products in cart
     */
    const productList = computed(() => {
      return Array.from(list.value.entries());
    });

    return { inTotal, productList };
  },
};
</script>
<style lang="scss" scoped>

.list{
    display: inline-grid;
    grid-template-columns: auto;
    margin-right: 2.8rem;
}
.list:after{
    content: "";
    display: block;
}

.inTotal {
    font-size: 1.3em;
}

</style>