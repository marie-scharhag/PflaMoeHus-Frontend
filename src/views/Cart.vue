<template>
    <div class="cart">
        <div id="carthead">
            <h3>Ihre Produkte</h3>
        </div>
        <div id="list">
            <CartList/>
        </div>
        <div id="order">
            <router-link :to="jwttoken.accessToken.length > 0  ? '/orderform'  : '/login'" append> <button id="checkout"> zur Kasse</button>  </router-link>
        </div>
  </div>
</template>
<script>
import CartList from "@/components/CartList.vue";
import "@/service/Product";
import { useUserStore } from "../service/UserStore";
import {routerHistory} from "../service/RouterStore";

export default {
  components: {
    CartList,
  },
  setup() {
    const { jwttoken } = useUserStore();
    routerHistory.add("/cart");
    return { jwttoken };
  },
};
</script>
<style scoped lang="scss">
.cart{
    display: flex;
    flex-direction: column;
} 
#list{
    display: flex;
    justify-content: center;
}

#carthead{
    display: flex;
    justify-content: center;
} 
#carthead h3{
    margin-top: 5%;
    text-align: center;
        
} 
#order{
    display:flex;
    justify-content: center;
} 

#checkout{
    padding: 20% 100%;
}

</style>