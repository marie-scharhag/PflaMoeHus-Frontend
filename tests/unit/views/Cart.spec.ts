import { shallowMount, mount, VueWrapper, RouterLinkStub } from "@vue/test-utils"
	
import Cart from "@/views/Cart.vue"
import CartList from "@/views/Cart.vue"
	
describe('Test CartList.vue', () => {
	
    let cmp: VueWrapper<any>
	
    beforeEach(() => {
        cmp = shallowMount(Cart)
    })
	
    test('has the expected html structur', () => {
        expect(cmp.element).toMatchSnapshot()
    })
	
	
    test('stubs a component using a custom component, component CartList is mentioned right in html', () => {

        const CartListMock = {	
            name: 'CartList',
            template: 'fakeCartList'
        }
        const wrapper = mount(Cart, {
            global: {
            stubs: { CartList: CartListMock }
            }
        }) 
        expect(wrapper.html()).toContain(`<div id=\"list\">fakeCartList</div>`) 
    })


    test('find out if Component CartList exists ', () => {
	
        const wrapper = shallowMount(Cart)
	
        wrapper.findComponent(CartList) // returns a VueWrapper
	
        expect(() => wrapper.getComponent('.not-there')).toThrowError()
    })
})