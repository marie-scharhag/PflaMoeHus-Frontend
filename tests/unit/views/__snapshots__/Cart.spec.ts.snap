// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CartList.vue has the expected html structur 1`] = `
<div
  class="cart"
>
  <div
    id="carthead"
  >
    <h3>
      Ihre Produkte
    </h3>
  </div>
  <div
    id="list"
  >
    <cart-list-stub />
  </div>
  <div
    id="order"
  >
    <router-link
      to="/login"
    >
      <button
        id="checkout"
      >
         zur Kasse
      </button>
    </router-link>
  </div>
</div>
`;
